modules/authfacebook/lib/Facebook.php

```diff
@@ -146,4 +146,11 @@ protected function constructSessionVariableName($key) {
     }
     return implode('_', $parts);
   }
+
+  protected function establishCSRFTokenState() {
```

Comment:
You're not using this function in your committed code??

Reply:
I'm overriding the behavior from base_facebook->establishCSRFTokenState
I suppose it could use a comment to describe why I'm overriding.

Reply:
Gah, sorry... It's too late to review code...
But don't we just need a newer version of the BaseFacebook-class??

Reply:
Get some sleep :) Facebook integrations won't break for 6 more weeks :)

I believe BaseFacebook was copied from https://github.com/facebookarchive/facebook-php-sdk/blob/master/src/base_facebook.php and that project has been deprecated. 
My quick read through the latest from that project was that it still wouldn't let us control the state query param.

As to the long term viability of the authfacebook module - my current thinking is to get rid of it, and build a generic Oauth2 auth module based on http://oauth2-client.thephpleague.com/ ( it would then support their 20+ social integrations ). I have an experimental version that works with Facebook and Amazon. I hope to finish it sometime this spring.

Reply:
Yes! I was reading the same and apparently this library is old and abandoned..

At this point, I feel all our social ID modules should move to separate repositories...
If we could replace them with one generic auth-module in the near future, that would be great!